### PR TITLE
MINOR: Add helper util `Snapshots.lastContainedLogTimestamp`

### PR DESCRIPTION
**Pull request by alok123t**
This PR refactors the `lastContainedLogTimestamp` to the Snapshots class, for reusability
- Introduces `IdentitySerde` based on ByteBuffer, required for using RecordsSnapshotReader
- As a result we also remove `recordSerde: RecordSerde[_]` argument from the `KafkaMetadataLog` constructor
- In addition, we also remove an unused variable in `SnapshotWriterReaderTest.java`

Testing:
Enhanced existing unit test `testWritingSnapshot` for testing

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/raft/KafkaMetadataLog.scala [hunk 1]
```diff
@@ -377,19 +375,7 @@ final class KafkaMetadataLog private (
    */
   private def readSnapshotTimestamp(snapshotId: OffsetAndEpoch): Option[Long] = {
     readSnapshot(snapshotId).asScala.map { reader =>
-      val recordsSnapshotReader = RecordsSnapshotReader.of(
-        reader,
-        recordSerde,
```
Comment (jsancio) [inline core/src/main/scala/kafka/raft/KafkaMetadataLog.scala:382]: I think you should be able to remove `recordSerde` from the constructor for `KafkaMetadataLog`.

File: raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java [hunk 2]
```diff
@@ -151,4 +155,24 @@ public static Path markForDelete(Path logDir, OffsetAndEpoch snapshotId) {
             );
         }
     }
+
+    public static long lastContainedLogTimestamp(RawSnapshotReader reader) {
+        try (RecordsSnapshotReader<ByteBuffer> recordsSnapshotReader =
+             RecordsSnapshotReader.of(
+                 reader,
+                 new IdentitySerde(),
```
Comment (jsancio) [inline raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java]: `RecordSerde` objects are reusable. You can add an `INSTANCE` static field to `IndentitySerde` and use it here. Take a look at `MetadataRecordSerde` for an example.

**Review by jsancio — COMMENTED** [review]
Thanks for the improvement @alok123t. Left you a few comments.

File: raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java (hunk 2 above)
Comment (alok123t) [reply raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java]: done in https://github.com/apache/kafka/pull/13772/commits/b10d32c85c9047b15d355239c44da62d9921bb6d

**Review by jsancio — APPROVED** [review]
LGTM

**Pull request merged**